src/tools/pkcs11-tool.c

```diff
@@ -6475,7 +6491,13 @@ static void show_dobj(CK_SESSION_HANDLE sess, CK_OBJECT_HANDLE obj)
 	printf("\n");
 
 	get_token_info(opt_slot, &info);
-	printf("  uri:            %s", get_uri(&info));
+	printf("  uri:            %s", get_uri(&info, opt_slot));
+	if ((id = getID(sess, obj, &idsize)) != NULL && idsize) {
+		printf(";id=");
+		for (unsigned int n = 0; n < idsize; n++)
+			printf("%%%02x", id[n]);
+		free(id);
+	}
```

Comment:
I think we discussed somewhere previously, that the Data objects do not have the CKA_ID attribute defined. So even if it would work for some custom implementations, it would result in false errors/warnings on compliant implementations, that I would like to avoid.

https://docs.oasis-open.org/pkcs11/pkcs11-spec/v3.1/os/pkcs11-spec-v3.1-os.html#_Toc111203218